isthmus/src/test/java/io/substrait/isthmus/api/TestIsthmusEndToEnd.java

```diff
@@ -0,0 +1,97 @@
+package io.substrait.isthmus.api;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.io.Resources;
+import io.substrait.extension.SimpleExtension;
+import io.substrait.isthmus.SqlToSubstrait;
+import io.substrait.isthmus.SubstraitToCalcite;
+import io.substrait.isthmus.SubstraitToSql;
+import io.substrait.isthmus.SubstraitTypeSystem;
+import io.substrait.plan.Plan.Root;
+import io.substrait.plan.ProtoPlanConverter;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.util.List;
+import org.apache.calcite.adapter.java.ReflectiveSchema;
+import org.apache.calcite.jdbc.JavaTypeFactoryImpl;
+import org.apache.calcite.schema.Schema;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.junit.jupiter.api.Test;
+
+/** API level testing */
+public class TestIsthmusEndToEnd {
+
+  /** Conversion of a Substrait Protobuf to */
+  @Test
+  public void substraitToSqlViaCalcite() throws IOException {
+
+    // create the protobuf substarit plan
+    byte[] planProtobuf = Resources.toByteArray(Resources.getResource("substrait_sql_003.plan"));
```

Comment:
other tests are using `SubstraitBuilder` to provide a Substrait plan for test cases which I think is more reliable than relying on a binary file in the tests resources which might have been encoded in an older version of the Substrait spec and which might break with future Substrait spec changes.

Reply:
fair point; will look at using that builder.